allennlp/data/dataset_readers/stanford_sentiment_tree_bank.py

```diff
@@ -0,0 +1,136 @@
+from typing import Dict, List
+import logging
+
+from overrides import overrides
+from nltk.tree import Tree
+import tqdm
+
+from allennlp.common import Params
+from allennlp.common.file_utils import cached_path
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import LabelField, TextField, Field
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import TokenIndexer, SingleIdTokenIndexer
+from allennlp.data.tokenizers import Token
+from allennlp.common.checks import ConfigurationError
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DatasetReader.register("sst_tokens")
+class StanfordSentimentTreeBankTokensDatasetReader(DatasetReader):
+    """
+    Reads tokens and their sentiment labels from the Stanford Sentiment Treebank.
+
+    Parameters
+    ----------
+    token_indexers : ``Dict[str, TokenIndexer]``, optional (default=``{"tokens": SingleIdTokenIndexer()}``)
+        We use this to define the input representation for the text.  See :class:`TokenIndexer`.
+        Note that the `output` tags will always correspond to single token IDs based on how they
+        are pre-tokenised in the data file.
+    use_subtrees : ``bool``, optional, (default = ``False``)
+        Whether or not to use sentiment-tagged subtrees.
+    granularity : ``str``, optional (default = ``"5-class"``)
+        One of ``"5-class"``, ``"3-class"``, or ``"2-class"``, indicating the number
+        of sentiment labels to use. The Stanford Sentiment Treebank comes with labels
+        from 0 to 4. ``"5-class"`` uses these labels as is. ``"3-class"`` converts the
+        problem into one of identifying whether a sentence is negative, positive, or
+        neutral sentiment. In this case, 0 and 1 are grouped as label 0 (negative sentiment),
+        2 is converted to label 1 (neutral sentiment) and 3 and 4 are grouped as label 2
+        (positive sentiment). ``"2-class"`` turns it into a binary classification problem
+        between positive and negative sentiment. 0 and 1 are grouped as the label 0
+        (negative sentiment), 2 (neutral) is discarded, and 3 and 4 are grouped as the label 1
+        (positive sentiment).
+    lazy : ``bool``, optional, (default = ``False``)
+        Whether or not instances can be consumed lazily.
```

Comment:
I think this is more about _reading_ instances lazily, not _consuming_ them lazily.